frontend/src/app/(routes)/(overview)/overview-components/OverviewPage.tsx

```diff
@@ -0,0 +1,292 @@
+  //             ?.coinMinimalDenom;
+  //         const decimals =
+  //           networks?.[chainID]?.network?.config?.currencies?.[0]?.coinDecimals ||
+  //           0;
+  //         totalRewards += convertToDollars(denom, rewards / 10 ** decimals);
+  //       }
+  //     });
+  //     return totalRewards;
+  //   }, [chainIDs, distributionChains, networks]);
+
+  const calculateTotalAvailableAmount = useCallback(() => {
+    let totalBalance = 0;
+    let totalIBCBalance = 0;
+
+    chainIDs.forEach((chainID) => {
+      const { minimalDenom, decimals, chainName } = getDenomInfo(chainID);
+      const balance = parseBalance(
+        balanceChains?.[chainID]?.list || [],
+        decimals,
+        minimalDenom
+      );
+
+      const ibcBalances = getIBCBalances(
+        balanceChains?.[chainID]?.list,
+        minimalDenom,
+        chainName
+      );
+
+      for (let i = 0; i < ibcBalances?.length; i++) {
+        totalIBCBalance += convertToDollars(
+          ibcBalances[i].balance.denom,
+          parseBalance(
+            [ibcBalances[i].balance],
+            ibcBalances?.[i]?.decimals,
+            ibcBalances?.[i]?.balance.denom
+          )
+        );
+      }
+
+      if (balanceChains?.[chainID]?.list?.length > 0) {
+        totalBalance += convertToDollars(minimalDenom, balance);
+      }
+    });
+
+    return { totalBalance, totalIBCBalance };
+  }, [chainIDs, balanceChains, getDenomInfo]);
+
+  const getSortedChainIds = useCallback(() => {
+    let sortedChains: { chainID: string; usdValue: number }[] = [];
+
+    chainIDs.forEach((chainID) => {
+      const { minimalDenom, decimals } = getDenomInfo(chainID);
+      const balanceAmountInDenoms = parseBalance(
+        balanceChains?.[chainID]?.list || [],
+        decimals,
+        minimalDenom
+      );
+
+      // minimalDenom
+      const stakedAmountInDenoms: number =
+        stakingChains?.[chainID]?.delegations?.totalStaked || 0;
+      // minimalDenom
+      // Todo: after distribution slice is added
+      //   const rewards =
+      //     distributionChains?.[chainID]?.delegatorRewards?.totalRewards || 0;
+      const rewardsAmountInDenoms: number = 0;
+      const chain: { chainID: string; usdValue: number } = {
+        chainID,
+        usdValue: 0,
+      };
+      if (balanceChains?.[chainID]?.list?.length > 0) {
+        chain.usdValue =
+          convertToDollars(minimalDenom, balanceAmountInDenoms) +
+          convertToDollars(
+            minimalDenom,
+            stakedAmountInDenoms / 10 ** decimals
+          ) +
+          convertToDollars(
+            minimalDenom,
+            rewardsAmountInDenoms / 10 ** decimals
+          );
+      }
+      sortedChains = [...sortedChains, chain];
+    });
+    sortedChains.sort((x, y) => y.usdValue - x.usdValue);
+    return sortedChains.map((chain) => chain.chainID);
+  }, [chainIDs, getDenomInfo, balanceChains]);
+
+  const getIBCSortedChainIds = useCallback(() => {
+    let sortedIBCChains: {
```

Comment:
Can we move all these parsing and transforming function into another file? Let's keep JSX pages small.